diff --git a/node_modules/react-native-v8/expo/src/main/java/io/csie/kudo/reactnative/v8/expo/V8ExpoAdapterPackage.java b/node_modules/react-native-v8/expo/src/main/java/io/csie/kudo/reactnative/v8/expo/V8ExpoAdapterPackage.java
index 012d716..fba4dee 100644
--- a/node_modules/react-native-v8/expo/src/main/java/io/csie/kudo/reactnative/v8/expo/V8ExpoAdapterPackage.java
+++ b/node_modules/react-native-v8/expo/src/main/java/io/csie/kudo/reactnative/v8/expo/V8ExpoAdapterPackage.java
@@ -28,7 +28,7 @@ public class V8ExpoAdapterPackage implements Package {
             @Nullable
             @Override
             public String getBundleAssetName(boolean useDeveloperSupport) {
-                final String v8BundleAssetName = V8ExecutorFactory.getBundleAssetName(context, useDeveloperSupport);
+                final String v8BundleAssetName = V8ExecutorFactory.getV8BundleAssetName(context, "v8codecache", useDeveloperSupport);
                 if (v8BundleAssetName != null) {
                     return v8BundleAssetName;
                 }
